server/service/client.go

```diff
@@ -2881,12 +2896,23 @@ func (c *Client) doGitOpsAndroidCertificates(config *spec.GitOps, logFn func(for
 	numCerts := len(certificates)
 
 	teamID := ""
+	teamName := ""
 	switch {
-	case config.TeamID != nil:
-		teamID = fmt.Sprintf("%d", *config.TeamID)
 	case config.IsNoTeam():
-		//  "No team"
+		teamName = "No team"
 		teamID = "0"
+	case config.TeamID != nil:
+		if config.TeamName != nil {
+			teamName = *config.TeamName
+		} else {
+			// Team name should already be provided in yml, but fetch it if not.
+			team, err := c.GetTeam(*config.TeamID)
+			if err != nil {
+				return fmt.Errorf("getting team by ID: %w", err)
+			}
+			teamName = team.Name
```

Comment:
I don't think I need to do a `GetTeam` call here.. because `name` is required in the team ymls. Policies and queries have team assigned directly in their yml definition:
```
policies:
  - name: My Policy
    team: Workstations  # ← Team name in YAML
    query: SELECT 1
```
android certs do not, but we can grab this from the name at the top level of the team yml.